pkg/auth/tokenparser.go

```diff
@@ -0,0 +1,92 @@
+package auth
+
+import (
+	"errors"
+	"log"
+
+	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+
+	jwt "github.com/dgrijalva/jwt-go"
+)
+
+// TokenClaims represents access token claims
+type TokenClaims struct {
+	Name          string         `json:"name"`
+	Username      string         `json:"preferred_username"`
+	GivenName     string         `json:"given_name"`
+	FamilyName    string         `json:"family_name"`
+	Email         string         `json:"email"`
+	EmailVerified bool           `json:"email_verified"`
+	Company       string         `json:"company"`
+	SessionState  string         `json:"session_state"`
+	Approved      bool           `json:"approved"`
+	Permissions   *[]Permissions `json:"permissions"`
+	jwt.StandardClaims
+}
+
+// Permissions represents a "permissions" claim in the AuthorizationPayload
+type Permissions struct {
```

Comment:
Do we need permissions here? This looks like some legacy code from `fabric8-auth`

Reply:
Ah, ok, thought that was something RHD specific.. Fixed 1d3f3aa.